app/helpers/application_helper.rb

```diff
@@ -22,11 +22,10 @@ def inbox_count
   def notification_count
     return 0 unless user_signed_in?
 
-    count = Notification.for(current_user).where(new: true)
-    return nil if count.nil?
-    return nil unless count.count.positive?
+    count = Notification.for(current_user).where(new: true).pluck(:id).size
```

Comment:
```suggestion
    count = Notification.for(current_user).where(new: true).count
```

would generate the nicer `COUNT(*)` query.  The resulting query takes around the same time as the one with pluck(:id).